src/VisualStudio/Core/Def/Workspace/VisualStudioSourceGeneratorTelemetryCollectorWorkspaceServiceFactory.cs

```diff
@@ -73,9 +73,9 @@ private void EnsureSubscribedToSolutionEvents()
         {
             Task.Run(async () =>
             {
-                var shellService = await _serviceProvider.GetServiceAsync<SVsSolution, IVsSolution>(_threadingContext.JoinableTaskFactory).ConfigureAwait(true);
+                var shellService = await _serviceProvider.GetServiceAsync<SVsSolution, IVsSolution>(throwOnFailure: true, _threadingContext.DisposalToken).ConfigureAwait(true);
                 await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(_threadingContext.DisposalToken);
-                shellService.AdviseSolutionEvents(this, out _);
+                shellService!.AdviseSolutionEvents(this, out _);
```

Comment:
I wonder if we should be putting the ! on the fetching so that way the nullable suppression is closer to the place where the throwOnFailure: true is. Or it's unfortunate if we can't get that nullable annotated in some way to make this simpler.

Reply:
Looks like there is a version of GetServiceAsync that only takes in a CancellationToken and assumes throwOnFailureTrue and has null annotated the return values. Switched over to use that when possible.